File: app/src/main/java/com/example/user/watched/pages/dashboard/di/DashboardBinderModule.java
package com.example.user.watched.pages.dashboard.di;

import com.example.user.watched.pages.dashboard.DashboardContract;
import com.example.user.watched.pages.dashboard.DashboardFragment;
import com.example.user.watched.pages.dashboard.DashboardPresenter;

import dagger.Binds;
import dagger.Module;

@Module
public interface DashboardBinderModule {


    @Binds
    DashboardContract.View bindDashboardFragment(DashboardFragment impl);



    @Binds
    DashboardContract.Presenter bindDashboardPresenter(DashboardPresenter impl);
}

File: app/src/main/java/com/example/user/watched/pages/dashboard/di/DashboardBinderModule.kt
package com.example.user.watched.pages.dashboard.di

import com.example.user.watched.pages.dashboard.DashboardContract
import com.example.user.watched.pages.dashboard.DashboardFragment
import com.example.user.watched.pages.dashboard.DashboardPresenter

import dagger.Binds
import dagger.Module

@Module
interface DashboardBinderModule {


    @Binds
    fun bindDashboardFragment(impl: DashboardFragment): DashboardContract.View


    @Binds
    fun bindDashboardPresenter(impl: DashboardPresenter): DashboardContract.Presenter
}
